### PR TITLE
Use Youtube atom

### PR DESCRIPTION
**Pull request by nicl**
# What does this change?

Nothing hopefully! But seriously, this change uses the `YoutubeAtom` to render `YoutubeBlockElement` elements, at least the bits that it can (Captions remain a DCR thing).

See also: https://github.com/guardian/atoms-rendering/pull/46.

### Before

Screenshot 2020-09-09 at 16 09 33

### After

Screenshot 2020-09-09 at 14 31 22

## Why?

To leverage `atoms-rendering` here.

### CODE REVIEW

File: src/lib/format.ts [hunk 1]
```diff
@@ -0,0 +1,85 @@
+import {
```
Comment (nicl) [inline src/lib/format.ts:1]: This file feels verbose, but it should be useful more generally as we migrate to `Format`.

File: src/web/components/elements/YoutubeBlockComponent.tsx [hunk 2]
```diff
@@ -64,26 +40,18 @@ export const YoutubeBlockComponent = ({
...
-                    />
-                )}
-                <iframe
-                    title={title}
-                    width={width}
-                    height={height}
-                    src={`https://www.youtube.com/embed/${element.assetId}?embed_config=${embedConfig}&enablejsapi=1&origin=https://www.theguardian.com&widgetid=1&modestbranding=1`}
-                />
-            </MaintainAspectRatio>
+            <YoutubeAtom
```
Comment (nicl) [inline src/web/components/elements/YoutubeBlockComponent.tsx:45]: (This is the key change for this PR.)

**Review by jfsoul — APPROVED** [review]

File: src/lib/content.d.ts [hunk 3]
```diff
@@ -324,8 +324,6 @@ interface YoutubeBlockElement {
     channelId?: string;
     duration?: number;
     posterSrc?: string;
-    height?: string;
-    width?: string;
```
[the remark this answers is not in the document]
Comment (nicl) [reply src/lib/content.d.ts:328]: @jfsoul they don't exist in the Frontend `YoutubeBlockElement` model so always be undefined. They are also unused in DCR too. I wonder if they were accidentally copied from the similar `VideoYoutubeBlockElement` but not sure to be honest.

File: scripts/webpack/browser.js [hunk 4]
```diff
@@ -74,7 +74,7 @@ module.exports = ({ isLegacyJS }) => ({
         rules: [
             {
                 test: /(\.tsx)|(\.js)|(\.ts)|(\.mjs)$/,
-                exclude: /node_modules\/(?!(@guardian\/discussion-rendering)|(dynamic-import-polyfill))\/.*/,
+                exclude: /node_modules\/(?!(@guardian\/discussion-rendering)|(@guardian\/types)|(dynamic-import-polyfill))\/.*/,
```
[the remark this answers is not in the document]
Comment (nicl) [reply scripts/webpack/browser.js:77]: True.

**Pull request merged**